src/Compilers/CSharp/Portable/Syntax/SyntaxFactory.cs

```diff
@@ -640,6 +642,492 @@ public static SyntaxToken XmlTextLiteral(SyntaxTriviaList leading, string text,
+        /// <param name="textTokens">A list of tokens used for the value of the xml text attribute.</param>
+        public static XmlTextAttributeSyntax XmlTextAttribute(XmlNameSyntax name, SyntaxKind quoteKind, SyntaxTokenList textTokens)
+        {
+            return XmlTextAttribute(
+                name,
+                Token(quoteKind),
+                textTokens,
+                Token(quoteKind))
+                .WithLeadingTrivia(Whitespace(" "));
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml element that spans multiple text lines.
+        /// </summary>
+        /// <param name="localName">The name of the xml element.</param>
+        /// <param name="content">A list of syntax nodes that represents the content of the xml multi line element.</param>
+        public static XmlElementSyntax XmlMultiLineElement(string localName, SyntaxList<XmlNodeSyntax> content)
+        {
+            return XmlMultiLineElement(XmlName(localName), content);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml element that spans multiple text lines.
+        /// </summary>
+        /// <param name="name">The name of the xml element.</param>
+        /// <param name="content">A list of syntax nodes that represents the content of the xml multi line element.</param>
+        public static XmlElementSyntax XmlMultiLineElement(XmlNameSyntax name, SyntaxList<XmlNodeSyntax> content)
+        {
+            return XmlElement(
+                XmlElementStartTag(name),
+                content.Insert(0, XmlNewLine()).Add(XmlNewLine()),
+                XmlElementEndTag(name));
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml text that contains a newline token with a documentation comment 
+        /// exterior trivia at the end (continued documentation comment).
+        /// </summary>
+        public static XmlTextSyntax XmlNewLine()
+        {
+            return XmlText(XmlTextNewLine());
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml newline token with a documentation comment exterior trivia at 
+        /// the end (continued documentation comment).
+        /// </summary>
+        public static SyntaxToken XmlTextNewLine()
+        {
+            return XmlTextNewLine(true);
+        }
+
+        /// <summary>
+        /// Creates the syntax representation of an xml newline token for xml documentation comments.
+        /// </summary>
+        /// <param name="continueXmlDocumentationComment">
+        /// If set to true, a documentation comment exterior token will be added to the trailing trivia
+        /// of the new token.</param>
+        public static SyntaxToken XmlTextNewLine(bool continueXmlDocumentationComment)
+        {
+            SyntaxToken token = XmlTextNewLine(
+                TriviaList(),
+                Environment.NewLine,
```

Comment:
No where else in the factory API do pickup platform behavior for new lines.  We have both kinds of new lines defined as separate properties on the factory.  The real issue here is that some of the other new API's use this new-line factory API. The benefit is that you get to use those API's without having to specify all the new line's manually.  The drawback, is that its picking the kind of new line for you without any means to control it.

The ways its done for the rest of the API is that no new lines are injected by any of the factory methods, only zero-width elastic whitespace.  We leave it up to the formatter to convert that elastic whitepace into new lines if the formatter deems them necessary.  That's probably the better way to handle it for these API's too. 

Note, the formatter doesn't yet have a solution to the platform new-line problem (such as when-in-Rome style), but it does operate using the value specified in the formatter new line option.